frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,353 @@
+/* eslint-disable no-undef */
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+
+import { connectToPlural } from '../../kinds';
+import { k8sCreate, K8sKind, K8sResourceKind, referenceFor } from '../../module/k8s';
+import { ButtonBar, history, LoadingBox, NameValueEditor, resourceObjPath, RequestSizeInput } from '../utils';
+import { RadioInput } from '../radio';
+// import { Checkbox } from '../checkbox';
+
+const Checkbox = ({ name, value, checked, onChange }) => (
+  <div>
+    <label className="checkbox-label">
+      <input type="checkbox" onChange={({ target: { checked: newChecked } }) => onChange(value, newChecked)} checked={!!checked} name={name} />
+      &nbsp;&nbsp;
+      {value}
+    </label>
+  </div>
+);
+
+// This will be commented out until ListDropdown gets added to master
+/*const StorageClassDropdown React.SFC<ScDropdownProps> = props => {
+
+  const kind = "StorageClass";
+  const resources = [{ kind }];
+  const selectedKey = props.selectedKey;
+  return (
+    <ListDropdown
+      {...props}
+      desc="Storage Classes"
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select storage class"
+      selectedKey={selectedKey}
+      required={required}
+      namespace={namespace}
+      name={name}
+    />
+  );
+};
+export type ScDropdownProps = {
+  namespace: string;
+  selectedKey: string;
+  required: boolean;
+  onChange: any;
+  id: string;
+  name: string;
+};*/
+
+export type PvcState = {
+  resourceObj: K8sResourceKind;
+  inProgress: boolean;
+  pvcName: string;
+  accessMode: string;
+  requestSizeValue: string;
+  requestSizeUnit: string;
+  defaultRequestSizeValue: string;
+  defaultRequestSizeUnit: string;
+  error?: any;
+  disableForm: boolean;
+  radios: any;
+  dropdownUnits: any;
+  showKeyValue: boolean;
+  nameValuePairs: any;
+};
+
+export type PvcProps = {
+  kindObj: K8sKind;
+  isCreate: boolean;
+  saveButtonText?: string;
+  namespace: string;
+};
+
+class CreatePvcForm extends React.Component<PvcProps, PvcState> {
+  constructor(props) {
+    super(props);
+    const resourceObj = {
+      apiVersion: 'v1',
+      kind: 'PersistentVolumeClaim',
+      metadata: {
+        name: '',
+        namespace: '',
+      },
+      spec: {
+        accessModes: {},
+        resources: {
+          requests: {
+            storage: '',
+          },
+        },
+        selector: {
+          matchLabels: {},
+        },
+      },
+    };
+
+    this.state = {
+      resourceObj: resourceObj,
+      inProgress: false,
+      pvcName: '',
+      accessMode: '',
+      requestSizeValue: '',
+      requestSizeUnit: '',
+      defaultRequestSizeValue: '1',
+      defaultRequestSizeUnit: 'Gi',
```

Comment:
Would remove `defaultRequestSizeValue` since I dont see any value in having a default value in the inputbox, would rather have the number `1` set as placeholder.
Also would remove `defaultRequestSizeUnit` and set the value directly to `requestSizeUnit`